data/test/scenarios/wml_tests/ScenarioWML/EventWML/events-test_filters.cfg

```diff
@@ -416,6 +416,250 @@
+                    [drains]
+                        id=ability_drain_blade
+                        [filter_opponent]
+                            [filter_weapon]
+                                special_id_active=ability_poison_blade
+                            [/filter_weapon]
+                        [/filter_opponent]
+                    [/drains]
+                [/abilities]
+            [/effect]
+            [filter]
+                id=alice
+            [/filter]
+        [/object]
+        [modify_unit]
+            [filter]
+            [/filter]
+            # Make sure they don't die during the attacks
+            [status]
+                invulnerable=yes
+            [/status]
+        [/modify_unit]
+        {VARIABLE triggers 0}
+        [do_command]
+            [move]
+                x=7,13
+                y=3,4
+            [/move]
+            [attack]
+                [source]
+                    x,y=13,4
+                [/source]
+                [destination]
+                    x,y=13,3
+                [/destination]
+            [/attack]
+        [/do_command]
+        [end_turn][/end_turn]
+    [/event]
+    [event]
+        name=side 2 turn
+        [do_command]
+            [attack]
+                [source]
+                    x,y=13,3
+                [/source]
+                [destination]
+                    x,y=13,4
+                [/destination]
+            [/attack]
+        [/do_command]
+        [end_turn][/end_turn]
+    [/event]
+    [event]
+        name=attack
+        first_time_only=no
+        [filter_attack]
+            special_id_active=ability_poison_blade
+        [/filter_attack]
+        {ASSERT ({VARIABLE_CONDITIONAL side_number equals 2})}
+        {VARIABLE_OP triggers_on_attack add 1}
+    [/event]
+    [event]
+        name=attack
+        first_time_only=no
+        [filter_second_attack]
+            special_id_active=ability_poison_blade
+        [/filter_second_attack]
+        {ASSERT ({VARIABLE_CONDITIONAL side_number equals 1})}
+        {VARIABLE_OP triggers_on_defense add 1}
+    [/event]
+    [event]
+        name=turn 2
+        {ASSERT ({VARIABLE_CONDITIONAL triggers_on_attack equals 1})}
+        {ASSERT ({VARIABLE_CONDITIONAL triggers_on_defense equals 1})}
+        {SUCCEED}
+    [/event]
+)}
+
+#####
+# API(s) being tested: [event][filter_attack],[event][filter_second_attack]
+##
+# Actions:
+# Define events filtering based on weapon specials.
+# Give Bob a poison special iff the opponent has a specific drain special.
+# Give Alice a drain special iff the opponent has a specific poison special.
+# Have side 1's unit attack side 2's unit.
+# Have side 2's unit attack side 1's unit.
+##
+# Expected end state:
+# Both events that shouldn't trigger because both special need of other for activate and cause infinite recursion, the specials remains inactive.
+#####
+{GENERIC_UNIT_TEST event_test_filter_attack_opponent_weapon_condition_no_triggered (
```

Comment:
This has ended up with two slightly different copies of event_test_filter_attack_opponent_weapon_condition_no_triggered. It also appears to be a test that could be simplified with COMMON_KEEP_A_B_UNIT_TEST.